sdk/src/test/java/io/opentelemetry/sdk/trace/export/BatchSampledSpansProcessorTest.java

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace.export;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.doThrow;
+
+import io.opentelemetry.proto.trace.v1.Span;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.TracerSdk;
+import io.opentelemetry.trace.util.Samplers;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+/** Unit tests for {@link BatchSampledSpansProcessor}. */
+@RunWith(JUnit4.class)
+public class BatchSampledSpansProcessorTest {
+  private static final String SPAN_NAME_1 = "MySpanName/1";
+  private static final String SPAN_NAME_2 = "MySpanName/2";
+  private static final long MAX_SCHEDULE_DELAY_MILLIS = 500;
+  private final TracerSdk tracerSdk = new TracerSdk();
+  private final WaitingSpanExporter waitingSpanExporter = new WaitingSpanExporter();
+  private final BlockingSpanExporter blockingSpanExporter = new BlockingSpanExporter();
+  @Mock private SpanExporter mockServiceHandler;
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @After
+  public void cleanup() {
+    tracerSdk.shutdown();
+  }
+
+  private ReadableSpan createSampledEndedSpan(String spanName) {
+    io.opentelemetry.trace.Span span =
+        tracerSdk.spanBuilder(spanName).setSampler(Samplers.alwaysSample()).startSpan();
+    span.end();
+    return (ReadableSpan) span;
+  }
+
+  private void createNotSampledEndedSpan(String spanName) {
```

Comment:
ditto

Reply:
Cannot be static because uses tracerSdk.